docs/adr/0007-allow-implementing-bindings-seperately.md

```diff
@@ -0,0 +1,98 @@
+# Allow implementing bindings in separate crates.
+
+* Status: Accepted
+* Deciders: teshaq, bdk, mhammond
+* Date: 2022-01-16
+
+Technical Story: [Issue 299](https://github.com/mozilla/uniffi-rs/issues/299)
+
+Implementation: [PR 1201](https://github.com/mozilla/uniffi-rs/pull/1201)
+
+Testing Implementation: [PR 1206](https://github.com/mozilla/uniffi-rs/pull/1206)
+
+## Context and Problem Statement
+All the binding generators currently live in the [`uniffi_bindgen`](../../uniffi_bindgen/src/bindings) crate. This creates the following difficulties:
+
+- All the bindings live in the `uniffi` repository, so the `uniffi` team has to maintain them (or at the very least review changes to them).
+- Any change to a specific binding generator requires a new `uniffi_bindgen` release for it to be accessible by consumers. Even if it doesn't impact any of the other bindings.
+- Some bindings require complex build systems to test. Including those build systems in `uniffi` would require developers to install those build systems, and CI to do the same. For example, any type of `gecko-js` bindings would require the mozilla-central build system to build and test.
+- We currently run all the tests for the bindings in our CI and through `cargo test`. This means that if one binding target gets outdated and fails, or if a developer doesn't have the needed libraries installed for one of the targets, the tests would fail.
+- It's currently difficult to support third-party developers writing bindings for languages the core team does not wish to maintain.
```

Comment:
for the sake of brevity, how about rolling this into the first point? ie, appending to that first point "this makes it difficult..."